quarkus/CONTRIBUTING.md

```diff
@@ -0,0 +1,211 @@
+
+## Running the server in development mode
+
+As a regular Quarkus application, you should be able to run the `server` module in [dev mode](https://quarkus.io/guides/maven-tooling#dev-mode) just like any regular application:
+
+```
+cd server
+mvn clean quarkus:dev -Dquarkus.args="start-dev"
+```
+
+You can set any command or configuration option to the server by setting the `quarkus.args` environment variable.
+
+When running in dev mode, you can benefit from the dev mode capabilities from Quarkus but with some limitations. The main limitations you'll find
+at the moment are:
+
+* Changes are only automatically reflected at runtime if you are changing resources from the `deployment`, `runtime`, and `server` modules. Other modules, such as `keycloak-services` still rely on Hot Swap in Java debuggers to reload classes.
+* There is nothing in the Dev UI related to the server itself, although you can still change some configuration from there.
+* There are some limitations when passing some options when running in dev mode. You should expect more improvements in this area.
+
+We are working to improve the dev experience, and you should expect improvements over time.
+
+## Debugging the server distribution
+
+The `kc.sh|bat` script allows you to remotely debug the distribution. For that, you should run the server as follows:
+
+```
+kc.sh --debug start-dev
+```
+
+By default, the debug port is available at `8787`.
+
+An additional environment variable `DEBUG_SUSPEND` can be set to suspend the JVM, when launched in debug mode. The `DEBUG_SUSPEND` variable supports the following values:
+
+* `y` - The debug mode JVM launch is suspended
+* `n` - The debug mode JVM is started without suspending
+
+Suspending the JVM when in debug mode is useful if you want to debug the early stages of the bootstrap code.
+
+When making changes to the `deployment`, `runtime`, or `server` modules, you can update the distribution with the new artifacts by executing
+the following command:
+
+```
+mvn -DskipTests clean install
+```
+
+After the `quarkus` module and sub-modules are built, you can update the distribution as follows:
+
+```
+cp -r server/target/lib ${KC_HOME_DIR}
+```
+
+In the example above, the `${KC_HOME_DIR}` variable points to the root directory of the distribution.
+
+You should also be able to update a server dependency directly. For that, copy the jar to the following location:
+
+```
+cp services/target/keycloak-services-${KC_VERSION}.jar ${KC_HOME_DIR}/lib/lib/main/org.keycloak.keycloak-services-${KC_VERSION}.jar
+```
+
+## Running tests
+
+The distribution has its own distribution and the main tests are within the [tests/integration](tests/integration) module.
+
+The test suite has two main types of tests:
+
+* `jvm`
+* `distribution`
+
+The `jvm` tests execute both the test class and server within the same JVM. While the `distribution` tests execute the server
+by running the distribution in a separte JVM.
```

Comment:
This is not 100% true because you can set `debug=true` to the annotation and start the server in debug mode. I see your point and if we want to go into details I would rather add how to debug distribution tests.

Reply:
works for me, too